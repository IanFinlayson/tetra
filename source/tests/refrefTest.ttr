def main():
   x = [1,2,3]
   print("Just created x and x = ", x, "\n")
   y = [[1,2,3],[4,5,6],[7,8,9]]
   print("Just created y and y = ", y, "\n")
   retVals = [0,0]
   print("Just created retVals and retVals = ", retVals, "\n")
   print("\n")
   print("\n")
   call1(x,retVals)
   print("Just returned from call1(x,retVals) now: \n")
   print("x = ", x, " and retVals = ", retVals, "\n")
   print("\n")
   call2(y,retVals)
   print("\n")
   print("Just returned from call2(y,retVals) now: \n")
   print("y = ", y, " and retVals = ", retVals, "\n")
   dd = [[[1]]]
   print("Just created dd and dd = ", dd, "\n")
   for i in dd:
      print("In outer for and i = ", i , "\n")
      for j in i:
         print("In inner for and j = ", j , "\n")
         j[0] = 123
         print("Just set j[0] = 123, now j = ", j, "\n")
      
   print("End of main and y[0][0] + dd[0][0][0] = ", y[0][0] + dd[0][0][0], "\n")

def call1(w [int],ret [int]):
   print("In call1 and w = ", w, " and ret = ", ret, "\n")
   for z in w:
      print("In for loop and z = ", z, "\n")
      z = z + 1
      print("Just incremented z and z = ", z, "\n")
   ret[0] = z
   print("Just set ret[0] = z and ret = ", ret, "\n")
   w[0] = 888
   print("Just set w[0] = 888 and w = ", w, "\n")

def call2(q [[int]],ret [int]):
   print("In call2 and q = ", q, " and ret = ", ret, "\n")
   for z in q:
      print("In outer for loop and z = ", z, "\n")
      for f in z:
         print("In inner for loop and f = ", f, "\n")
         f += 1
         print("Just incremented f and f = ", f, "\n")
      print("Calling call1(z, ret) from call2().\n")
      print("\n")
      call1(z,ret)
      print("\n")
      print("Just returned from call1(x,retVals) to call2() now: \n")
      print("z = ", z, " and ret = ", ret, "\n")
      z[0] = 99
      print("Just set z[0] = 99 and z = ", z, "\n")
   ret[1] = q[2][0]
   print("Just set ret[1] = q[2][0] and ret = ", ret, "\n")
