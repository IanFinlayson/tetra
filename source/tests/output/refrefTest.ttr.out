Just created x and x = [1, 2, 3]
Just created y and y = [[1, 2, 3], [4, 5, 6], [7, 8, 9]]
Just created retVals and retVals = [0, 0]


In call1 and w = [1, 2, 3] and ret = [0, 0]
In for loop and z = 1
Just incremented z and z = 2
In for loop and z = 2
Just incremented z and z = 3
In for loop and z = 3
Just incremented z and z = 4
Just set ret[0] = z and ret = [4, 0]
Just set w[0] = 888 and w = [888, 2, 3]
Just returned from call1(x,retVals) now: 
x = [888, 2, 3] and retVals = [4, 0]

In call2 and q = [[1, 2, 3], [4, 5, 6], [7, 8, 9]] and ret = [4, 0]
In outer for loop and z = [1, 2, 3]
In inner for loop and f = 1
Just incremented f and f = 2
In inner for loop and f = 2
Just incremented f and f = 3
In inner for loop and f = 3
Just incremented f and f = 4
Calling call1(z, ret) from call2().

In call1 and w = [1, 2, 3] and ret = [4, 0]
In for loop and z = 1
Just incremented z and z = 2
In for loop and z = 2
Just incremented z and z = 3
In for loop and z = 3
Just incremented z and z = 4
Just set ret[0] = z and ret = [4, 0]
Just set w[0] = 888 and w = [888, 2, 3]

Just returned from call1(x,retVals) to call2() now: 
z = [888, 2, 3] and ret = [4, 0]
Just set z[0] = 99 and z = [99, 2, 3]
In outer for loop and z = [4, 5, 6]
In inner for loop and f = 4
Just incremented f and f = 5
In inner for loop and f = 5
Just incremented f and f = 6
In inner for loop and f = 6
Just incremented f and f = 7
Calling call1(z, ret) from call2().

In call1 and w = [4, 5, 6] and ret = [4, 0]
In for loop and z = 4
Just incremented z and z = 5
In for loop and z = 5
Just incremented z and z = 6
In for loop and z = 6
Just incremented z and z = 7
Just set ret[0] = z and ret = [7, 0]
Just set w[0] = 888 and w = [888, 5, 6]

Just returned from call1(x,retVals) to call2() now: 
z = [888, 5, 6] and ret = [7, 0]
Just set z[0] = 99 and z = [99, 5, 6]
In outer for loop and z = [7, 8, 9]
In inner for loop and f = 7
Just incremented f and f = 8
In inner for loop and f = 8
Just incremented f and f = 9
In inner for loop and f = 9
Just incremented f and f = 10
Calling call1(z, ret) from call2().

In call1 and w = [7, 8, 9] and ret = [7, 0]
In for loop and z = 7
Just incremented z and z = 8
In for loop and z = 8
Just incremented z and z = 9
In for loop and z = 9
Just incremented z and z = 10
Just set ret[0] = z and ret = [10, 0]
Just set w[0] = 888 and w = [888, 8, 9]

Just returned from call1(x,retVals) to call2() now: 
z = [888, 8, 9] and ret = [10, 0]
Just set z[0] = 99 and z = [99, 8, 9]
Just set ret[1] = q[2][0] and ret = [10, 99]

Just returned from call2(y,retVals) now: 
y = [[99, 2, 3], [99, 5, 6], [99, 8, 9]] and retVals = [10, 99]
Just created dd and dd = [[[1]]]
In outer for and i = [[1]]
In inner for and j = [1]
Just set j[0] = 123, now j = [123]
End of main and y[0][0] + dd[0][0][0] = 222

